plugins/scaffolder-backend/src/service/rules.ts

```diff
@@ -129,6 +130,29 @@ function buildHasProperty<Schema extends z.ZodType<JsonPrimitive>>({
   });
 }
 
+export const createTaskPermissionRule = makeCreatePermissionRule<
+  string,
+  {},
+  typeof RESOURCE_TYPE_SCAFFOLDER_TASK
+>();
+
+export const isTaskOwner = createTaskPermissionRule({
+  name: 'IS_TASK_OWNER',
+  description: 'Allows tasks to be accessed by their creator',
+  resourceType: RESOURCE_TYPE_SCAFFOLDER_TASK,
+  paramsSchema: z.object({
+    createdBy: z.string().describe('Creator of the scaffolder task'),
+  }),
+  apply: (taskOwner, { createdBy }) => {
```

Comment:
following the `PermissionRule`'s suggestion above, this would be `task` an the logic of the function should be adjusted accordingly. 

```suggestion
  apply: (task, { createdBy }) => {
```

Reply:
As discussed above, I made it a string as to match on the filter so that the check is made before the database query is made. Then if the specified user is allowed to user the filter, they then use the built-in `createdBy` filter to grab the users.

Reply:
I noticed that recently a new change was added to the `/v2/tasks` endpoint where it now accepts filters that might include a list of `createdBy` `userEntityRef`s. Would this be for filtering for tasks owned by a list of users? If so, I can update my PR for that case as well since the original purpose of this PR was restrict users to only tasks they own.